### PR TITLE
Return empty Config fields, now omitempty, for API < 1.21

### LINKED ISSUE #17131: 1.9: Config.MacAddress gets dropped
In Docker 1.8 if you didn't send a mac address on create `Config.MacAddress` in inspect would equal empty string.  In 1.9 the field is dropped if it is the empty value.  For backwards compatibility this field should not be dropped but be empty string.

### LINKED ISSUE #17139: 1.9: Config.NetworkDisabled gets dropped
If `HostConfig.NetworkDisabled` is false the field is dropped.  This was not the case in Docker <=1.8.

### LINKED ISSUE #17173: 1.9: Config.ExposedPorts is being dropped
`ExposedPorts: null` would be sent in the inspect in Docker 1.8 but 1.9 is dropping the field

### PR DESCRIPTION
**Pull request by runcom**
- Fix #17131
- Fix #17139
- Fix #17173

Signed-off-by: Antonio Murdaca runcom@redhat.com

### CODE REVIEW

File: integration-cli/docker_api_inspect_test.go [hunk 1]
```diff
@@ -133,3 +133,34 @@ func (s *DockerSuite) TestInspectApiImageResponse(c *check.C) {
...
+	cleanedContainerID := strings.TrimSpace(out)
+
+	cases := []string{"1.19", "1.20"}
+	for _, version := range cases {
+		endpoint := fmt.Sprintf("/v%s/containers/%s/json", version, cleanedContainerID)
+		status, body, err := sockRequest("GET", endpoint, nil)
+		c.Assert(err, check.IsNil)
+		c.Assert(status, check.Equals, http.StatusOK)
+
+		var inspectJSON map[string]interface{}
```
Comment (tonistiigi) [inline integration-cli/docker_api_inspect_test.go]: Bit simpler code if you use: `var inspectJSON struct{ Config map[string]*json.RawMessage }`

**Comment by tonistiigi** [comment]
Do we need something similar for images as well, or is containers enough?

**Comment by runcom** [comment]
images don't contain Config fields (except for Labels) in API output

**Comment by tonistiigi** [comment]
@runcom `GET /images/name/json` contains `Config` and `ContainerConfig` that are both typed `runconfig.Config`. Seems to me that these show the same content that is on the disk without transforming anything.

**Comment by runcom** [comment]
@tonistiigi yes I was looking at `Image` for /images/json, fixing

**Comment by runcom** [comment]
funny thing is we never versioned image type for inspect output

**Comment by runcom** [comment]
ping @calavera do we want to have bc layers for v1.20 and pre1.20 for images as well? If yes I'll add it to this PR since I've already worked on it

**Comment by calavera** [comment]
@runcom yes, we should be as much backwards compatible as possible. On the other hand, we haven't had reports of breaking changes regarding images, so I could be okay not having it for 1.9.

@tiborvass what do you think?

**Comment by tiborvass** [comment]
+1, let's merge this as-is for 1.9

**Comment by vdemeester** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by cpuguy83** [comment]
Running these tests locally now.

**Comment by cpuguy83** [comment]
Tests pass locally. merging.

**Pull request merged**